code/game/objects/structures/flora.dm

```diff
@@ -53,6 +50,21 @@
 	name = "xmas tree"
 	desc = "A wondrous decorated Christmas tree."
 	icon_state = "pine_c"
+	var/gifts_under_tree = FALSE
+	var/list/ckeys_that_took = list()
+
+/obj/structure/flora/tree/pine/xmas/attack_hand(mob/living/user)
+	if(!gifts_under_tree)
+		return
+	if(!user.ckey)
+		return
+	if(user.ckey in ckeys_that_took)
+		to_chat(user, "<span class='warning'>You already took your gift.</span>")
+		return
+	to_chat(user, "<span class='warning'>After a bit of rummaging, you locate a gift with your name on it!</span>")
+	ckeys_that_took += user.ckey

```

Comment:
`ckeys_that_took[user.ckey] = TRUE`
then to check:
`if(ckeys_that_took[user.ckey])`